### PR TITLE
[feature](cloud) Support path prefix for HDFS storage vault.

### PR DESCRIPTION
**Pull request by wangshuo128**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/util/vault_info_util.cpp
Comment (gavinchou) [inline be/src/util/vault_info_util.cpp:1]: seems no need to add a new file.
this helper function can be inlined in cloud_storage_engin.cpp

File: be/src/util/vault_info_util.cpp [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "vault_info_util.h"
+
+namespace doris {
+
+THdfsParams to_hdfs_params(const cloud::HdfsVaultInfo& vault) {
```
Comment (gavinchou) [inline be/src/util/vault_info_util.cpp:22]: also consider renaming HdfsVaultInfo to HdfsVaultInfoPB to keep the name consistent.
maybe in a separate PR

File: be/src/util/vault_info_util.cpp
Comment (wangshuo128) [reply be/src/util/vault_info_util.cpp:1]: I think this converting logic should also be used by `recycler`.

Comment (wangshuo128) [reply be/src/util/vault_info_util.cpp:1]: After checking, `recycler` and `BE` are different projects...
moving this to cloud_storage_engin.cpp.

File: be/src/cloud/cloud_meta_mgr.cpp [hunk 2]
```diff
@@ -826,18 +826,7 @@ Status CloudMetaMgr::get_storage_vault_info(
...
-        params.hdfs_kerberos_keytab = vault.hdfs_info().build_conf().hdfs_kerberos_keytab();
-        params.hdfs_kerberos_principal = vault.hdfs_info().build_conf().hdfs_kerberos_principal();
-        for (const auto& confs : vault.hdfs_info().build_conf().hdfs_confs()) {
-            THdfsConf conf;
-            conf.key = confs.key();
-            conf.value = confs.value();
-            params.hdfs_conf.emplace_back(std::move(conf));
-        }
-        vault_infos->emplace_back(vault.id(), std::move(params));
+        vault_infos->emplace_back(vault.id(), std::move(vault.hdfs_info()));
```
Comment (platoneko) [inline be/src/cloud/cloud_meta_mgr.cpp:829]:
```suggestion
        if (vault.has_hdfs_info()) {
            vault_infos->emplace_back(vault.id(), vault.hdfs_info());
        }
```
std::move has no effect on const

File: be/src/util/vault_info_util.cpp
Comment (platoneko) [reply be/src/util/vault_info_util.cpp:1]: Consider to move these code to `io/hdfs_util.cpp`

File: be/src/cloud/cloud_meta_mgr.h [hunk 3]
```diff
@@ -71,7 +71,7 @@ class CloudMetaMgr {
     Status precommit_txn(const StreamLoadContext& ctx);
 
     Status get_storage_vault_info(
```
Comment (platoneko) [inline be/src/cloud/cloud_meta_mgr.h:73]:
```suggestion
    using StorageVaultInfos = std::vector<...>;
    Result<StorageVaultInfos> get_storage_vault_info()
```

**Review by platoneko — COMMENTED** [review]
Almost LGTM

**Comment by wangshuo128** [comment]
run buildall

**Comment by wangshuo128** [comment]
run buildall

**Comment by wangshuo128** [comment]
run buildall

**Review by ByteYue — APPROVED** [review]

**Comment by wangshuo128** [comment]
run buildall

**Comment by wangshuo128** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by gavinchou — APPROVED** [review]

**Pull request merged**